packages/formatter-markdown/src/formatter.ts

```diff
@@ -0,0 +1,224 @@
+import { logger } from '@hint/utils';
+
+import { FormatterOptions, HintResources, IFormatter } from 'hint';
+import { Problem, Category, Severity } from '@hint/utils-types';
+import AnalysisResult, { CategoryResult, HintResult } from './result';
+
+import * as fs from 'fs-extra';
+
+import path = require('path');
+import { cwd } from 'process';
+import { MarkdownHelpers, HeaderCount } from './utils';
+import { getMessage as getMessageFormatter, MessageName } from './i18n.import';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Utils
+ * ------------------------------------------------------------------------------
+ */
+
+const messagesFileName = 'messages.json';
+
+/* istanbul ignore next */
+const getCategoryListFromResources = (resources: HintResources) => {
+    const categoriesArray: string[] = resources.hints.map((hint) => {
+        if (hint.meta.docs && hint.meta.docs.category) {
+            return hint.meta.docs.category;
+        }
+
+        return Category.other;
+    });
+
+    // Clean duplicated values.
+    const categories: Set<string> = new Set(categoriesArray);
+
+    return Array.from(categories);
+};
+
+const getCategoryList = (resources?: HintResources): string[] => {
+    /* istanbul ignore if */
+    if (resources) {
+        return getCategoryListFromResources(resources);
+    }
+
+    const result: string[] = [];
+
+    for (const [, value] of Object.entries(Category)) {
+        result.push(value);
+    }
+
+    return result;
+};
+
+const createLanguageFile = async (language: string = 'en') => {
+    const rootPath = path.join(__dirname, '_locales');
+    const languagesToCheck = [language];
+    const languageParts = language.split('-');
+
+    /*
+     * Add to the list the 'main' language.
+     * e.g. en-US => en
+     */
+    if (languageParts.length > 1) {
+        languagesToCheck.push(languageParts[0]);
+    }
+
+    // Default to 'en'.
+    let existingLanguage = 'en';
+
+    for (const lang of languagesToCheck) {
+        const file = path.join(rootPath, lang, messagesFileName);
+
+        if (fs.existsSync(file)) { // eslint-disable-line no-sync
```

Comment:
Can you please add a comment saying `fs.exists` is deprecated and that's why we use the sync version?